comp/core/log/params.go

```diff
@@ -0,0 +1,149 @@
+	params.loggerName = loggerName
+	if overrideFromEnv {
+		params.logLevelFn = func(configGetter) string { return config.GetEnvDefault("DD_LOG_LEVEL", level) }
+	} else {
+		params.logLevelFn = func(configGetter) string { return level }
+	}
+	params.logFileFn = func(configGetter) string { return "" }
+	params.logSyslogURIFn = func(configGetter) string { return "" }
+	params.logSyslogRFCFn = func(configGetter) bool { return false }
+	params.logToConsoleFn = func(configGetter) bool { return true }
+	params.logFormatJSONFn = func(configGetter) bool { return false }
+	return params
+}
+
+// LogForDaemon sets up logging parameters for a daemon app.
+//
+// The log level is set based on the `log_level` config parameter.
+//
+// The log file is set based on the logFileConfig config parameter,
+// or disabled if `disable_file_logging` is set.
+//
+// On platforms which support it, syslog is enabled if `log_to_syslog` is set,
+// using `syslog_uri` or defaulting to "unixgram:///dev/log" if that is empty.
+// The `syslog_rfc` config parameter determines whether this produces 5424-compliant
+// output.
+//
+// Console logging is enabled if `log_to_console` is set.  Lots are formatted
+// as JSON if `log_format_json` is set.
+func LogForDaemon(loggerName, logFileConfig, defaultLogFile string) Params {
+	params := Params{}
+	params.loggerName = loggerName
+	params.logLevelFn = func(g configGetter) string { return g.GetString("log_level") }
+	params.logFileFn = func(g configGetter) string {
+		if g.GetBool("disable_file_logging") {
+			return ""
+		}
+		logFile := g.GetString(logFileConfig)
+		if logFile == "" {
+			logFile = defaultLogFile
+		}
+		return logFile
+	}
+	params.logSyslogURIFn = func(g configGetter) string {
+		if runtime.GOOS == "windows" {
+			return "" // syslog not supported on Windows
+		}
+		enabled := g.GetBool("log_to_syslog")
+		uri := g.GetString("syslog_uri")
+
+		if !enabled {
+			return ""
+		}
+
+		if uri == "" {
+			return "unixgram:///dev/log"
+		}
+
+		return uri
+	}
+	params.logSyslogRFCFn = func(g configGetter) bool { return g.GetBool("syslog_rfc") }
+	params.logToConsoleFn = func(g configGetter) bool { return g.GetBool("log_to_console") }
+	params.logFormatJSONFn = func(g configGetter) bool { return g.GetBool("log_format_json") }
+	return params
+}
+
+// LogToFile modifies the parameters to set the destination log file, overriding any
+// previous logfile parameter.
+func (params *Params) LogToFile(logFile string) {
+	params.logFileFn = func(configGetter) string { return logFile }
+}
+
+// These functions are used in unit tests.
+
+// LogLevelFn returns the log level
+func (params Params) LogLevelFn(c configGetter) string {
+	return params.logLevelFn(c)
+}
+
+// LogFileFn returns the log file
+func (params Params) LogFileFn(c configGetter) string {
+	return params.logFileFn(c)
+}
+
+// LoggerName is the name that appears in the logfile
+func (params Params) LoggerName() string {
+	return params.loggerName
+}
```

Comment:
I think we should align those with `config` and also use the `With` prefix. WDYT ?

Reply:
The last 3 methods are getters and there is no reason to use `With` prefix for them.

`LogForOneShot` and `LogForDaemon` are constructors for `log.Params` so no reason to prefix with `With`.

`LogToFile` can be a valid candidate for `With` but I am not sure it worths to use the option pattern for a single field (and `LogToFile` is used once in the code). What do you think?

Reply:
Sorry, I miss that the last 3 were getters. I think it's worth it for `LogToFile`, to at least highlight that it's not a getter like the other ones.